ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js

```diff
@@ -39,173 +40,226 @@ const HandleBarsHelpers = require('../../js/HandlebarsHelpers.js')
 const ResultLinkView = require('../result-link/result-link.view.js')
 const plugin = require('plugins/result-item')
 
+const LIST_DISPLAY_TYPE = 'List'
+const GRID_DISPLAY_TYPE = 'Grid'
+
+const getResultDisplayType = () =>
+  (user &&
+    user
+      .get('user')
+      .get('preferences')
+      .get('resultDisplay')) ||
+  LIST_DISPLAY_TYPE
+
+class ResultIndicatorContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { isSelected: props.model.get('isSelected') }
+  }
+
+  componentDidMount = () => {
+    const updateIsSelected = model => {
+      const isSelected = model.get('isSelected')
+      this.setState({ isSelected: isSelected })
+    }
+
+    this.props.model.once('change', updateIsSelected)
+  }
+
+  componentWillUnmount = () => this.props.model.stopListening()
+
+  render = () => {
+    return (
+      !this.state.isSelected && (
+        <MarionetteRegionContainer
+          className="container-indicator"
+          view={ResultIndicatorView}
+          viewOptions={{ model: this.props.model }}
+        />
+      )
+    )
+  }
+}
+
 const ResultItemView = Marionette.LayoutView.extend({
   template(data) {
+    const model = this.model
+    const displayAsGrid = getResultDisplayType() === GRID_DISPLAY_TYPE
+    const renderThumbnail = displayAsGrid && data.metacard.properties.thumbnail
+    const IndicatorContext = React.createContext()
+
     return (
       <React.Fragment>
-        <div
-          className="result-container"
-          data-metacard-id={data.id}
-          data-query-id={data.metacard.queryId}
-        >
-          <div className="container-indicator" />
-          <div className="container-content">
-            <div className="content-header">
-              <span
-                className="header-icon fa fa-history"
-                title="Type: Revision"
-                data-help="Indicates the type of result
+        <IndicatorContext.Provider value={model}>
```

Comment:
I don't think we should be using context here.

Reply:
Why is that? This prevents having to call `render` on model changes.

Reply:
When the model changes, you do want to rerender though right? The bigger issue for me is that context is designed to solve a different problem. A problem of passing information down a view tree without intermediate components being explicitly aware. This doesn't look like the problem you are solving.

Reply:
This is exactly what I'm trying to accomplish, i.e. not re-rendering the `result-item.view` but the indicator container https://github.com/codice/ddf/pull/4310/files#diff-675f3fe2a9461cf5a8d35c38b729298aR71